### PR TITLE
Marketplace: Fixes search bar on the header when scrolling.

### PR DESCRIPTION
**Pull request by Rolilink**
#### Screencast
https://user-images.githubusercontent.com/1035546/162019892-a216fb03-4e45-44ad-925d-c1191a9f1f43.mov

#### Changes proposed in this Pull Request

* Fixes search bar to the header when scrolling down.

#### Testing instructions
* Go to `/plugins/:siteId`
* Scroll down pass popular searches it should stick to the top.
* Scroll up pass popular searches it should render the search box in the previous position.
* On mobile devices it shouldn't stick the search bar to the header.

Related to #62229

### CODE REVIEW

File: client/my-sites/plugins/search-box-header/index.jsx [hunk 1]
```diff
@@ -15,21 +15,22 @@ const SearchBox = ( { isMobile, doSearch, searchTerm, searchBoxRef } ) => {
...
+				ref={ searchBoxRef }
 				pinned={ isMobile }
 				fitsContainer={ isMobile }
 				onSearch={ doSearch }
 				defaultValue={ searchTerm }
 				placeholder={ translate( 'Try searching "ecommerce"' ) }
 				delaySearch={ true }
 				recordEvent={ recordSearchEvent }
-				ref={ searchBoxRef }
+				searchMode="on-enter"
```
Comment (lsl) [inline client/my-sites/plugins/search-box-header/index.jsx]: Can we move this change to another PR? I'm a little wary of assuming everyone knows enter = submit ([we hide the submit button](https://github.com/Automattic/wp-calypso/issues/62540)) .

Comment (Rolilink) [reply client/my-sites/plugins/search-box-header/index.jsx]: @lsl I think this is actually not a submit button, [it seems to be a button used to open the search bar on mobile](https://github.com/Automattic/wp-calypso/blob/trunk/packages/search/src/search.tsx#L395)

Comment (Rolilink) [reply client/my-sites/plugins/search-box-header/index.jsx]: but divided this one into two PRs for testing reasons

File: client/my-sites/plugins/plugins-browser/index.jsx [hunk 2]
```diff
@@ -119,6 +120,12 @@ const PluginsBrowser = ( {
 	hideHeader,
 	doSearch,
 } ) => {
+	const {
+		isAboveElement,
+		targetRef: searchHeaderRef,
+		referenceRef: navigationHeaderRef,
+	} = useScrollAboveElement();
```
Comment (paulopmt1) [inline client/my-sites/plugins/plugins-browser/index.jsx:127]: Should we consider replace our custom `use-scroll-above-element` to the [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)?

It's pretty reliable right now: 

image

**Review by paulopmt1 — APPROVED** [review]
Good job @Rolilink!

After your change, I can successfully use it in my desktop browsers: Firefox, Chrome, and Edge.
One optional suggestion only.

File: client/my-sites/plugins/plugins-browser/index.jsx (hunk 2 above)
Comment (Rolilink) [reply client/my-sites/plugins/plugins-browser/index.jsx:127]: This is a good follow-up for 20% time 👍🏼  thanks @paulopmt1 !
